Source/WebCore/Modules/cache/CacheStorageConnection.h

```diff
@@ -59,8 +63,10 @@ class CacheStorageConnection : public ThreadSafeRefCounted<CacheStorageConnectio
     uint64_t computeRecordBodySize(const FetchResponse&, const DOMCacheEngine::ResponseBody&);
 
     // Used only for testing purposes.
-    virtual void clearMemoryRepresentation(const ClientOrigin&, DOMCacheEngine::CompletionCallback&& callback) { callback(DOMCacheEngine::Error::NotImplemented); }
-    virtual void engineRepresentation(CompletionHandler<void(const String&)>&& callback) { callback(String { }); }
+    using CompletionPromise = NativePromise<void, DOMCacheEngine::Error>;
```

Comment:
I would use a `GenericPromise` here instead, considering the error code is ignored in the end, and we only care if it resolved or rejected.